storage/feed.go

```diff
@@ -255,6 +257,39 @@ func (s *Storage) fetchFeedsWithCounters(query string, args ...interface{}) (mod
 	return feeds, nil
 }
 
+// FeedSchedulerParameters returns the parameters used for the scheduler.
+func (s *Storage) FeedSchedulerParameters(userID, feedID int64) (int, error) {
+	scheduler := strings.ToLower(config.Opts.PollingScheduler())
+	if scheduler != model.SchedulerEntryCountBased {
+		return 0, nil
+	}
+	var weeklyCount int
+
+	query := `
+		SELECT
+			(SELECT count(*) FROM entries 
+				WHERE entries.feed_id=f.id AND 
+				(entries.published_at Between (now() - interval '7 days') and now()) 
+				) as weekly_count
+		FROM feeds f
+		WHERE
+			f.user_id=$1 AND f.id=$2
```

Comment:
This query could be simplified: 

```
SELECT
	count(*)
FROM
	entries
WHERE
	entries.user_id=$1 AND 
	entries.feed_id=$2 AND 
	entries.published_at BETWEEN (now() - interval '1 week') AND now();
```

And this index could be added to your SQL migration:

```
create index entries_user_feed_idx on entries (user_id, feed_id);
```

Reply:
done. thanks.
could you tell me more about how the index helps speeding up the query?

Reply:
If you do a explain analyze, you will see the index used:

```
explain analyze SELECT count(*) FROM entries WHERE entries.user_id=1234 AND entries.feed_id=4242 AND entries.published_at BETWEEN (now() - interval '1 week') AND now();
                                                              QUERY PLAN
--------------------------------------------------------------------------------------------------------------------------------------
 Aggregate  (cost=8.46..8.47 rows=1 width=8) (actual time=0.801..0.802 rows=1 loops=1)
   ->  Index Scan using entries_user_feed_idx on entries  (cost=0.43..8.46 rows=1 width=0) (actual time=0.517..0.774 rows=63 loops=1)
         Index Cond: ((user_id = 1234) AND (feed_id = 4242))
         Filter: ((published_at <= now()) AND (published_at >= (now() - '7 days'::interval)))
         Rows Removed by Filter: 119
 Planning Time: 0.592 ms
 Execution Time: 0.946 ms
```

The same query without index:

```
explain analyze SELECT count(*) FROM entries WHERE entries.user_id=1234 AND entries.feed_id=4242 AND entries.published_at BETWEEN (now() - interval '1 week') AND now();
                                                                      QUERY PLAN
------------------------------------------------------------------------------------------------------------------------------------------------------
 Aggregate  (cost=171.44..171.45 rows=1 width=8) (actual time=10.686..10.687 rows=1 loops=1)
   ->  Bitmap Heap Scan on entries  (cost=167.41..171.44 rows=1 width=0) (actual time=10.591..10.669 rows=63 loops=1)
         Recheck Cond: ((feed_id = 4242) AND (user_id = 1234))
         Filter: ((published_at <= now()) AND (published_at >= (now() - '7 days'::interval)))
         Rows Removed by Filter: 119
         Heap Blocks: exact=127
         ->  BitmapAnd  (cost=167.41..167.41 rows=1 width=0) (actual time=10.346..10.347 rows=0 loops=1)
               ->  Bitmap Index Scan on entries_feed_id_hash_key  (cost=0.00..13.81 rows=168 width=0) (actual time=0.132..0.133 rows=182 loops=1)
                     Index Cond: (feed_id = 4242)
               ->  Bitmap Index Scan on entries_user_status_idx  (cost=0.00..153.35 rows=8123 width=0) (actual time=10.175..10.175 rows=7985 loops=1)
                     Index Cond: (user_id = 55)
 Planning Time: 0.403 ms
 Execution Time: 10.765 ms
```